CoreScriptsRoot/CoreScripts/NotificationScript2.lua

```diff
@@ -0,0 +1,472 @@
+end
+
+removeNotification = function(notification)
+	if not notification then return end
+	--
+	local index = findNotification(notification)
+	table.remove(NotificationQueue, index)
+	local frame = notification.Frame
+	if frame and frame.Parent then
+		frame:TweenPosition(UDim2.new(1, 0, 1, frame.Position.Y.Offset), EASE_DIR, EASE_STYLE, TWEEN_TIME, true,
+			function()
+				frame:Destroy()
+			end)
+	end
+	if #OverflowQueue > 0 then
+		local nextNofication = OverflowQueue[1]
+		table.remove(OverflowQueue, 1)
+		insertNotifcation(nextNofication)
+	else
+		updateNotifications()
+	end
+end
+
+local function sendNotifcation(title, text, image, duration, callback)
+	local notification = {}
+	local notificationFrame = createNotification(title, text, image)
+	--
+	notification.Frame = notificationFrame
+	notification.Duration = duration
+	insertNotifcation(notification)
+end
+BindableEvent_SendNotification.OnInvoke = function(title, text, image, duration, callback)
+	sendNotifcation(title, text, image, duration, callback)
+end
+
+local function sendFriendNotification(fromPlayer)
+	local notification = {}
+	local notificationFrame = createNotification(fromPlayer.Name, "Sent you a friend request!",
+		FRIEND_IMAGE..tostring(fromPlayer.userId).."&x=48&y=48")
+	--
+	local acceptButton = createTextButton("AcceptButton", "Accept", UDim2.new(0, 0, 1, 2))
+	acceptButton.Parent = notificationFrame
+
+	local declineButton = createTextButton("DeclineButton", "Decline", UDim2.new(0.5, 2, 1, 2))
+	declineButton.Parent = notificationFrame
+
+	acceptButton.MouseButton1Click:connect(function()
+		if notification then
+			removeNotification(notification)
+		end
+		LocalPlayer:RequestFriendship(fromPlayer)
+	end)
+
+	declineButton.MouseButton1Click:connect(function()
+		if notification then
+			removeNotification(notification)
+		end
+		LocalPlayer:RevokeFriendship(fromPlayer)
+		FriendRequestBlacklist[fromPlayer] = true
+	end)
+
+	notification.Frame = notificationFrame
+	notification.Duration = 8
+	notification.IsFriend = true
+	insertNotifcation(notification)
+end
+
+--[[ Friend Notifications ]]--
+local function onFriendStatusChanged(otherPlayer, status)
+	if status == Enum.FriendStatus.FriendRequestReceived then
+		if FriendRequestBlacklist[otherPlayer] then return end
+		sendFriendNotification(otherPlayer)
+	elseif status == Enum.FriendStatus.Friend then
+		sendNotifcation("New Friend", "You are now friends with "..otherPlayer.Name.."!", 
+			FRIEND_IMAGE..tostring(otherPlayer.userId).."&x=48&y=48", DEFAULT_NOTIFICATION_DURATION, nil)
+	end
+end
+LocalPlayer.FriendStatusChanged:connect(onFriendStatusChanged)
+
+--[[ Player Points Notifications ]]--
+local time = 20
+local function onPointsAwarded(userId, pointsAwarded, userBalanceInGame, userTotalBalance)
+	--if userId == LocalPlayer.userId then
+		if pointsAwarded == 1 then
+			sendNotifcation("Point Awarded", "You received "..tostring(time).." point!", PLAYER_POINTS_IMG, DEFAULT_NOTIFICATION_DURATION, nil)
```

Comment:
Yep, it was left in from a test case I was running, to test variable time notifications. Will be updated.